enterprise_access/apps/api/v1/tests/test_subsidy_access_policy_views.py

```diff
@@ -23,15 +29,309 @@
     PerLearnerEnrollmentCapLearnerCreditAccessPolicyFactory,
     PerLearnerSpendCapLearnerCreditAccessPolicyFactory
 )
-
-from .utils import BaseEnterpriseAccessTestCase
+from test_utils import APITestWithMocks
 
 SUBSIDY_ACCESS_POLICY_LIST_ENDPOINT = reverse('api:v1:policy-list')
 SUBSIDY_ACCESS_POLICY_ADMIN_LIST_ENDPOINT = reverse('api:v1:admin-policy-list')
 
+TEST_ENTERPRISE_UUID = uuid4()
+
+
+# pylint: disable=missing-function-docstring
+class CRUDViewTestMixin:
+    """
+    Mixin to set some basic state for test classes that cover the
+    subsidy access policy CRUD views.
+    """
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+
+        cls.enterprise_uuid = TEST_ENTERPRISE_UUID
+
+        cls.redeemable_policy = PerLearnerEnrollmentCapLearnerCreditAccessPolicyFactory(
+            enterprise_customer_uuid=cls.enterprise_uuid,
+            spend_limit=3,
+            active=True,
+        )
+        cls.non_redeemable_policy = PerLearnerEnrollmentCapLearnerCreditAccessPolicyFactory(
+            enterprise_customer_uuid=cls.enterprise_uuid,
+            spend_limit=0,
+            active=True,
+        )
+
+    def setUp(self):
+        super().setUp()
+        # Start in an unauthenticated state.
+        self.client.logout()
+
+
+@ddt.ddt
+class TestPolicyCRUDAuthNAndPermissionChecks(CRUDViewTestMixin, APITestWithMocks):
+    """
+    Tests Authentication and Permission checking for Subsidy Access Policy CRUD views.
+    """
+    @ddt.data(
+        # A role that's not mapped to any feature perms will get you a 403.
+        (
+            {'system_wide_role': 'some-other-role', 'context': str(TEST_ENTERPRISE_UUID)},
+            status.HTTP_403_FORBIDDEN,
+        ),
+        # A good admin role, but in a context/customer we don't have, get's you a 403.
+        (
+            {'system_wide_role': SYSTEM_ENTERPRISE_ADMIN_ROLE, 'context': str(uuid4())},
+            status.HTTP_403_FORBIDDEN,
+        ),
+        # A learner role is also fine, but in a context/customer we don't have, get's you a 403.
+        (
+            {'system_wide_role': SYSTEM_ENTERPRISE_LEARNER_ROLE, 'context': str(uuid4())},
+            status.HTTP_403_FORBIDDEN,
+        ),
+        # An operator role is fine, too, but in a context/customer we don't have, get's you a 403.
```

Comment:
if the customer is missing - should we offer a 404 instead? makes reasoning client-side easier?